docs/develop/plugins-guide.mdx

```diff
@@ -0,0 +1,351 @@
+
+- AI Agent SDKs
+- Observability, tracing, or logging middleware
+- Adding reliable built-in functionality such as LLM calls, corporate messaging, and payments infrastructure
+- Encryption or compliance middleware
+
+## How to build a Plugin
+
+The recommended way to start building plugins is with a `SimplePlugin`. This abstraction will tackle the vast majority of plugins people want to write.  
+
+For advanced use cases, you can extend the methods in lower-level classes that Simple Plugin is based on without re-implementing what you’ve done. See the [Advanced Topics section](#advanced-topics-for-plugins) for more information.
+
+### Example Plugins
+
+If you prefer to learn by getting hands-on with code, check out some existing plugins.
+
+- Temporal's Python SDK ships with an [OpenAI Agents SDK](https://github.com/temporalio/sdk-python/tree/main/temporalio/contrib/openai_agents) plugin
+- Pydantic plugin when it’s fixed
+
+## What you can provide to users in a plugin
+
+There are a number of features you can give your users with a plugin. Here's a short list of some of the things you can do and more are being added.
+
+- [Built-in Activities](#built-in-activity)
+- [Workflow libraries](#workflow-libraries)
+- [Built-in Child Workflows](#built-in-child-workflows)
+- [Built-in Nexus Operations](#built-in-nexus-operations)
+- [Custom Data Converters](#custom-data-converters)
+- [Interceptors](#interceptors)
+
+### Built-in Activity
+
+You can provide built-in Activities in a Plugin for users to call from their Workflows. Activities are the most common Temporal primitive and should be scoped to:
+
+- Single write operations
+- Batches of similar writes
+- One or more read operations followed by a write operation
+- A read that should be memoized, like an LLM call, a large download, or a slow-polling read
+
+Larger pieces of functionality should be broken it up into multiple activities. This makes it easier to do failure recovery, have short timeouts, and be idempotent.
+
+Here are some best practices you can use when you are making Activity plugins:
+
+- Activity arguments and return values must be serializable.
+- Activities that perform writes should be idempotent.
+- Activities have [timeouts](https://docs.temporal.io/develop/python/failure-detection#heartbeat-timeout) and [retry policies](https://docs.temporal.io/encyclopedia/retry-policies). To be Activity-friendly, your operation should either complete within a few minutes or it should support the ability to heartbeat or poll for a result. This way it will be clear to the Workflow when the Activity is still making progress.
+- You need to specify at least one timeout, typically the [start_to_close_timeout](https://docs.temporal.io/encyclopedia/detecting-activity-failures#start-to-close-timeout). Keep in mind that the shorter the timeout, the faster Temporal will retry upon failure. See the [timeouts and retry policies](#timeouts-and-retry-policies) to learn more.
```

Comment:
Definitely seems like we should link to general advice on activities somewhere in this section at least